exec/execPack.sh

```diff
@@ -65,28 +74,32 @@ build_ami() {
   echo "building AMI"
   echo "-----------------------------------"
 
-  packer build -machine-readable -var aws_access_key=$aws_access_key_id \
-    -var aws_secret_key=$aws_secret_access_key \
-    -var REGION=$REGION \
-    -var VPC_ID=$VPC_ID \
-    -var SUBNET_ID=$SUBNET_ID \
-    -var SECURITY_GROUP_ID=$SECURITY_GROUP_ID \
-    -var AMI_ID=$AMI_ID \
-    -var IMAGE_NAMES_SPACED="${IMAGE_NAMES_SPACED}" \
-    -var REL_VER=$RES_REL_VER_NAME \
-    -var REL_DASH_VER=$RES_REL_VER_NAME_DASH \
-    execAMI.json 2>&1 | tee output.txt
-
-  #this is to get the ami from output
-  echo versionName=$(cat output.txt | awk -F, '$0 ~/artifact,0,id/ {print $6}' \
-  | cut -d':' -f 2) > "$JOB_STATE/$CURR_JOB.env"
-
-  echo "RES_REL_VER_NAME=$RES_REL_VER_NAME" >> "$JOB_STATE/$CURR_JOB.env"
-  echo "RES_REL_VER_NAME_DASH=$RES_REL_VER_NAME_DASH" >> "$JOB_STATE/$CURR_JOB.env"
-  echo "RES_IMG_VER_NAME=$RES_IMG_VER_NAME" >> "$JOB_STATE/$CURR_JOB.env"
-  echo "RES_IMG_VER_NAME_DASH=$RES_IMG_VER_NAME_DASH" >> "$JOB_STATE/$CURR_JOB.env"
-  echo "IMAGE_NAMES_SPACED=$IMAGE_NAMES_SPACED" >> "$JOB_STATE/$CURR_JOB.env"
-  cat "$JOB_STATE/$CURR_JOB.env"
+  if [ "$RES_REL_VER_NAME" == "$DRYDOCK_REL_VER_NAME" ]; then
+    packer build -machine-readable -var aws_access_key=$aws_access_key_id \
+      -var aws_secret_key=$aws_secret_access_key \
+      -var REGION=$REGION \
+      -var VPC_ID=$VPC_ID \
+      -var SUBNET_ID=$SUBNET_ID \
+      -var SECURITY_GROUP_ID=$SECURITY_GROUP_ID \
+      -var AMI_ID=$AMI_ID \
+      -var IMAGE_NAMES_SPACED="${IMAGE_NAMES_SPACED}" \
+      -var REL_VER=$RES_REL_VER_NAME \
+      -var REL_DASH_VER=$RES_REL_VER_NAME_DASH \
+      execAMI.json 2>&1 | tee output.txt
+
+    #this is to get the ami from output
+    echo versionName=$(cat output.txt | awk -F, '$0 ~/artifact,0,id/ {print $6}' \
+    | cut -d':' -f 2) > "$JOB_STATE/$CURR_JOB.env"
+
+    echo "RES_REL_VER_NAME=$RES_REL_VER_NAME" >> "$JOB_STATE/$CURR_JOB.env"
+    echo "RES_REL_VER_NAME_DASH=$RES_REL_VER_NAME_DASH" >> "$JOB_STATE/$CURR_JOB.env"
+    echo "RES_IMG_VER_NAME=$RES_IMG_VER_NAME" >> "$JOB_STATE/$CURR_JOB.env"
+    echo "RES_IMG_VER_NAME_DASH=$RES_IMG_VER_NAME_DASH" >> "$JOB_STATE/$CURR_JOB.env"
+    echo "IMAGE_NAMES_SPACED=$IMAGE_NAMES_SPACED" >> "$JOB_STATE/$CURR_JOB.env"
+    cat "$JOB_STATE/$CURR_JOB.env"
+  else
+    echo "SHIPPABLE_RELEASE not same as DRYDOCK_RELEASE, skipping Machine Image creation"
```

Comment:
Should we preserve the last known state of the job (especially `versionName`)? It would be strange for this job to be successful with empty state.

Reply:
makes sense. updated the script to save versionName in job state